src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/GCRefMapBuilder.cs

```diff
@@ -290,6 +289,26 @@ private void GcScanValueType(TypeDesc type, ArgDestination argDest, int delta, C
             }
         }
 
+        private void FindByRefPointerOffsetsInByRefLikeObject(TypeDesc type, ArgDestination argDest, int delta, CORCOMPILE_GCREFMAP_TOKENS[] frame)
+        {
+            if (type.IsByRefLike)
```

Comment:
This condition does not look right

Reply:
Thanks for spotting, I have also realized it was incorrect. I have pushed a second commit that should be hopefully fixing it.